viscy/representation/engine.py

```diff
@@ -133,20 +163,46 @@ def validation_step(self, batch: TripletSample, batch_idx: int) -> Tensor:
         """Validation step of the model."""
         anchor = batch["anchor"]
         pos_img = batch["positive"]
-        neg_img = batch["negative"]
         anchor_projection = self(anchor)
-        negative_projection = self(neg_img)
         positive_projection = self(pos_img)
-        loss = self.loss_function(
-            anchor_projection, positive_projection, negative_projection
-        )
-        self._log_metrics(
-            loss, anchor_projection, positive_projection, negative_projection, "val"
-        )
-        if batch_idx < self.log_batches_per_epoch:
-            self.validation_step_outputs.extend(
-                detach_sample((anchor, pos_img, neg_img), self.log_samples_per_batch)
+        if isinstance(self.loss_function, NTXentLoss):
+            indices = torch.arange(
+                0, anchor_projection.size(0), device=anchor_projection.device
+            )
+            labels = torch.cat((indices, indices))
+            # Note: we assume the two augmented views are the anchor and positive samples
+            embeddings = torch.cat((anchor_projection, positive_projection))
+            loss = self.loss_function(embeddings, labels)
+            self._log_metrics(
+                loss=loss,
+                anchor=anchor_projection,
+                positive=positive_projection,
+                negative=None,
+                stage="val",
             )
+            if batch_idx < self.log_batches_per_epoch:
+                self.validation_step_outputs.extend(
+                    detach_sample((anchor, pos_img), self.log_samples_per_batch)
+                )
+        else:
+            neg_img = batch["negative"]
+            negative_projection = self(neg_img)
+            loss = self.loss_function(
+                anchor_projection, positive_projection, negative_projection
+            )
+            self._log_metrics(
+                loss=loss,
+                anchor=anchor_projection,
+                positive=positive_projection,
+                negative=negative_projection,
+                stage="val",
+            )
+            if batch_idx < self.log_batches_per_epoch:
+                self.validation_step_outputs.extend(
+                    detach_sample(
+                        (anchor, pos_img, neg_img), self.log_samples_per_batch
+                    )
+                )
```

Comment:
Both his branch and the same branch in `training_step` have some repetition with the NT-Xent branches. Is it possible to abstract out some code?